services/bls_aggregation/blsagg.go

```diff
@@ -0,0 +1,364 @@
+package blsagg
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"math/big"
+	"sync"
+	"time"
+
+	"github.com/Layr-Labs/eigensdk-go/crypto/bls"
+	"github.com/Layr-Labs/eigensdk-go/logging"
+	"github.com/Layr-Labs/eigensdk-go/services/avsregistry"
+	"github.com/Layr-Labs/eigensdk-go/types"
+)
+
+var (
+	TaskAlreadyInitializedErrorF = func(taskIndex types.TaskIndex) error {
+		return fmt.Errorf("task %d already initialized", taskIndex)
+	}
+	TaskExpiredError   = fmt.Errorf("task expired")
+	TaskNotFoundErrorF = func(taskIndex types.TaskIndex) error {
+		return fmt.Errorf("task %d not initialized or already completed", taskIndex)
+	}
+	OperatorNotPartOfTaskQuorumErrorF = func(operatorId types.OperatorId, taskIndex types.TaskIndex) error {
+		return fmt.Errorf("operator %x not part of task %d's quorum", operatorId, taskIndex)
+	}
+	SignatureVerificationError = func(err error) error {
+		return fmt.Errorf("Failed to verify signature: %w", err)
+	}
+	IncorrectSignatureError = errors.New("Signature verification failed. Incorrect Signature.")
+)
+
+// BlsAggregationServiceResponse is the response from the bls aggregation service
+// it's half of the data needed to build the NonSignerStakesAndSignature struct
+type BlsAggregationServiceResponse struct {
+	Err                          error
+	TaskIndex                    types.TaskIndex
+	TaskResponseDigest           types.TaskResponseDigest
+	NonSignersPubkeysG1          []*bls.G1Point
+	QuorumApksG1                 []*bls.G1Point
+	SignersApkG2                 *bls.G2Point
+	SignersAggSigG1              *bls.Signature
+	NonSignerQuorumBitmapIndices []uint32
+	QuorumApkIndices             []uint32
+	TotalStakeIndices            []uint32
+	NonSignerStakeIndices        [][]uint32
+}
+
+// aggregatedOperators is meant to be used as a value in a map
+// map[taskResponseDigest]aggregatedOperators
+type aggregatedOperators struct {
+	// aggregate g2 pubkey of all operatos who signed on this taskResponseDigest
+	signersApkG2 *bls.G2Point
+	// aggregate signature of all operators who signed on this taskResponseDigest
+	signersAggSigG1 *bls.Signature
+	// aggregate stake of all operators who signed on this header for each quorum
+	signersTotalStakePerQuorum map[types.QuorumNum]*big.Int
+	// OperatorId of operators who signed on this header for each quorum
+	signersOperatorIdsSet map[types.OperatorId]bool
```

Comment:
the type `map[types.OperatorId]bool` is only a map from an operatorID to a bool. There are no multiple quorums here.

Reply:
thanks, fixed.